src/components/form/textInput/index.js

```diff
@@ -41,7 +41,7 @@ const TextInput = ({ value, onChange, name, label, icon, type = "text" }) => {
 		return (
 			<div className="inputwrapper">
 				{label && <label htmlFor={name}>{label}</label>}
-				<input type={type} name={name} value={value} onChange={onChange} className={icon && 'input-has-icon'} />
+				<input type={type} name={name} value={value} onChange={onChange} className={icon && 'input-has-icon'} readOnly={readOnly || false} />
```

Comment:
`readonly || false` will not be needed if `readOnly` has a default value of `false`